lib/SPIRV/libSPIRV/SPIRVDecorate.h

```diff
@@ -78,7 +78,9 @@ class SPIRVDecorateGeneric : public SPIRVAnnotationGeneric {
   void setOwner(SPIRVDecorationGroup *Owner) { this->Owner = Owner; }
 
   SPIRVCapVec getRequiredCapability() const override {
-    return getCapability(Dec);
+    auto Caps = getAllEnablingCapabilities(Dec);
+    Module->chooseBestCapability(Caps);
```

Comment:
```suggestion
    Module->chooseBestCapability(getAllEnablingCapabilities(Dec));
```

Reply:
There is "return Caps" one line below, so we cannot optimize it this way

Reply:
Oh, completely missed that, sorry